From 1ef1096641607e0b74bd60784d87afd4e88d0cc3 Mon Sep 17 00:00:00 2001
From: "Steven J. Hill" <Steven.Hill@cavium.com>
Date: Thu, 16 May 2019 16:48:28 +0200
Subject: [PATCH 288/345] EDAC: octeon_edac-lmc: Fix module removal when ECC
 unsupported.

Signed-off-by: Steven J. Hill <steven.hill@cavium.com>
---
 drivers/edac/octeon_edac-lmc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/edac/octeon_edac-lmc.c b/drivers/edac/octeon_edac-lmc.c
index ab04b0e7fbf3..d2d71fd47d5c 100644
--- a/drivers/edac/octeon_edac-lmc.c
+++ b/drivers/edac/octeon_edac-lmc.c
@@ -343,8 +343,11 @@ static int octeon_lmc_edac_remove(struct platform_device *pdev)
 {
 	struct mem_ctl_info *mci = platform_get_drvdata(pdev);
 
+	if (mci == NULL)
+		goto out;
 	edac_mc_del_mc(&pdev->dev);
 	edac_mc_free(mci);
+out:
 	return 0;
 }
 
-- 
2.25.1

