From 4f07c693d402cd16b8b119d8ca856db3a1c130a7 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Thu, 9 May 2019 18:19:50 +0200
Subject: [PATCH 292/345] rtc: isl12026: Fix build failure when CONFIG_NVMEM
 not enabled.

Signed-off-by: David Daney <david.daney@cavium.com>
---
 drivers/rtc/rtc-isl12026.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/rtc/rtc-isl12026.c b/drivers/rtc/rtc-isl12026.c
index 693315fa4982..69bc7de512d7 100644
--- a/drivers/rtc/rtc-isl12026.c
+++ b/drivers/rtc/rtc-isl12026.c
@@ -5,6 +5,7 @@
  * Copyright (c) 2018 Cavium, Inc.
  */
 #include <linux/types.h>
+#include <linux/err.h>
 #include <linux/nvmem-provider.h>
 #include <linux/of_device.h>
 #include <linux/module.h>
-- 
2.25.1

