From 6c4aac1bf0fc9c52dbdd727b4671639967f317e7 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Sat, 8 Jun 2019 15:34:33 +0200
Subject: [PATCH 316/345] MIPS/EDAC: Call edac handle for bigrd/bigwd cases.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
---
 drivers/edac/octeon_edac-l2c.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/edac/octeon_edac-l2c.c b/drivers/edac/octeon_edac-l2c.c
index f2471a845d9a..398d82b696fb 100644
--- a/drivers/edac/octeon_edac-l2c.c
+++ b/drivers/edac/octeon_edac-l2c.c
@@ -272,11 +272,13 @@ static void _octeon_l2c_poll_oct3(struct edac_device_ctl_info *l2c, int tad)
 		snprintf(buf1, sizeof(buf1),
 			"Read reference past L2C_BIG_CTL[MAXDRAM] occurred:");
 		l2c_reset.cn70xx.bigrd = true;
+		edac_device_handle_ce(l2c, tad, 0, buf1);
 	}
 	if (l2c_err.s.bigwr) {
 		snprintf(buf1, sizeof(buf1),
 			"Write reference past L2C_BIG_CTL[MAXDRAM] occurred:");
 		l2c_reset.cn70xx.bigwr = true;
+		edac_device_handle_ce(l2c, tad, 0, buf1);
 	}
 
 	if (l2c_reset.u64)
-- 
2.25.1

